src/app/domain/backup/services/manageProgress.ts

```diff
@@ -0,0 +1,66 @@
+import Minilog from '@cozy/minilog'
+
+import CozyClient, { generateWebLink } from 'cozy-client'
+import { getNativeIntentService } from 'cozy-intent'
+
+import { BackupInfo } from '/app/domain/backup/models'
+
+const log = Minilog('💿 Backup')
+
+const getPhotosUri = (client: CozyClient): string => {
+  const cozyUrl = client.getStackClient().uri
+  const subDomainType = client.getInstanceOptions().capabilities.flat_subdomains
+    ? 'flat'
+    : 'nested'
+
+  const photosUri = generateWebLink({
+    cozyUrl,
+    subDomainType,
+    slug: 'photos',
+    pathname: '',
+    hash: '',
+    searchParams: []
+  })
+
+  return photosUri
+}
+
+const getHomeUri = (client: CozyClient): string => {
+  const cozyUrl = client.getStackClient().uri
+  const subDomainType = client.getInstanceOptions().capabilities.flat_subdomains
+    ? 'flat'
+    : 'nested'
+
+  const homeUri = generateWebLink({
+    cozyUrl,
+    subDomainType,
+    slug: 'home',
+    pathname: '',
+    hash: '',
+    searchParams: []
+  })
+
+  return homeUri
+}
+
+export const sendProgressToWebview = (
+  client: CozyClient,
+  backupInfo: BackupInfo
+): void => {
+  const nativeIntentService = getNativeIntentService()
+
+  const photosUri = getPhotosUri(client)
+  const homeUri = getHomeUri(client)
+
+  nativeIntentService
+    .call(photosUri, 'updateBackupInfo', backupInfo)
+    ?.catch((e: Error) =>
+      log.debug('Error when sending backup info to photos app', e.message)
+    )
+
+  nativeIntentService
+    .call(homeUri, 'updateBackupInfo', backupInfo)
+    ?.catch((e: Error) =>
+      log.debug('Error when sending backup info to home app', e.message)
```

Comment:
My understanding is that we expect to always have one of Home or Photo errors to occur? If yes maybe we should mention that in the log. Something like 'expected error if alone'.

Reply:
Updated this code to use a try catch which is more really readable.

Home or Photo **may** not be here. If yes, _nativeIntentService.call_ return _undefined_. So in this case, I had to add the _?.catch_ to avoid call catch on _undefined_

With a try / catch

```
  try {
    await nativeIntentService.call(photosUri, 'updateBackupInfo', backupInfo)
  } catch (e) {
    if (e instanceof Error) {
      log.debug('Error when sending backup info to photos app', e.message)
    }
  }
```